Add a Dockerfile

LiquidPL opened the pull request:
As mentioned in sio2project/oioioi#126, in this PR I'm adding a `Dockerfile` that builds a working Docker image for sioworkers. It might also be worth adding a GitHub Actions workflow that builds a new image after creating a new release, but I would prefer to consult with project maintainers on where the images should be published beforehand.

I've also noticed that there's an old Docker setup in `docker/`, however since it seems fairly old, I wasn't sure if it's being in active use, and if there's a point in updating it.

A-dead-pixel commented:
I think there should also be a way to set the concurrency and memory limit via environment variables.

twalen commented on README.md:
```diff
@@ -1,27 +1,98 @@
 
-### for python 2 installation ###
-pip install -r requirements.txt
+## Python 2
```
is there any need to support Python2?

LiquidPL replied:
I'm not sure. The project supported Python 2 beforehand so I didn't remove it. Besides, at least in my testing, sio2jail doesn't seem work on Python 3 right now (see #11), so I assumed there must be some Python 2 based deployments somewhere.

If there is no need to keep Python 2 working, then I can remove it.

LiquidPL commented:
I've cleaned up the readme file and `setup.py` file from references to Python 2. The actual code will still need to be refactored to target Python 3 specifically.

Additionally, I've added workflows to run the tests on every push, and to publish a Docker image on every new release (we need to decide where to publish said images, though).

LiquidPL commented:
I have updated the build action to push images to Docker Hub.

For it to work, the Docker Hub username and password/[access token](https://docs.docker.com/docker-hub/access-tokens/) need to be configured as secrets in the [repository settings](https://github.com/sio/sioworkers/settings/secrets/actions):

* `REGISTRY_USERNAME` holds the username,
* `REGISTRY_PASSWORD` holds the password/token.

Docker's documentation recommends creating an access token for the purpose of pushing images.

Additionally, a `sio2project/sioworkers` repository needs to be created in Docker Hub.
